### PR TITLE
Added UART assisted test RPI scripts. Updated 

### PR DESCRIPTION
**Pull request by lundinc2**
Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by qiutongs** [comment]
Did you changed some file permission for libraries/abstractions/common_io/test/test_scripts/uart/__init__.py

File: libraries/abstractions/common_io/test/test_iot_uart.c [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_NOT_EQUAL( ultestIotUartFlowControl, pvOriginalConfig.ucFlowControl );
+        TEST_ASSERT_NOT_EQUAL( ultestIotUartFlowControl, pvgetConfigBuffer.ucFlowControl );
+
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartGetConfig, &pvgetConfigBuffer );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_EQUAL( ultestIotUartFlowControl, pvgetConfigBuffer.ucFlowControl );
+
+        /* Test changing stop bit config */
+        pvConfigBuffer.xStopbits = ultestIotUartStopBits;
```
Comment (cmcgcc) [inline libraries/abstractions/common_io/test/test_iot_uart.c]: In the previous test_iot_uart.c, there is a xSampleConfig2 and is used here. It has different configuration. Why is it taken away?

File: libraries/abstractions/common_io/test/test_iot_uart.c [hunk 2]
```diff
@@ -0,0 +1,817 @@
...
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
+    if( TEST_PROTECT() )
+    {
+
+        /* Save original UART config */
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartGetConfig, &pvOriginalConfig );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_NOT_EQUAL( testIotUART_TEST_BAUD_RATE, pvOriginalConfig.ulBaudrate );
```
Comment (cmcgcc) [inline libraries/abstractions/common_io/test/test_iot_uart.c]: Why the original config is checked here? How can you ensure it is with the default values?

File: libraries/abstractions/common_io/test/test_iot_uart.c (hunk 1 above)
Comment (lundinc2) [reply libraries/abstractions/common_io/test/test_iot_uart.c]: This PR is about a month old, and test_iot_uart.c  has had some changes and been rebased. I think we should scrap this PR and rebase the python scripts onto the latest version.

File: libraries/abstractions/common_io/test/test_iot_uart.c [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartSetConfig, &pvOriginalConfig );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+
+        /* Getting transfer byte amount only works on an operation in flight. */
+        lIoctl = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount );
+        TEST_ASSERT_EQUAL( IOT_UART_INVALID_VALUE, lIoctl );
+
+        lIoctl = iot_uart_ioctl( xUartHandle, eGetRxNoOfbytes, &lTransferAmount );
+        TEST_ASSERT_EQUAL( IOT_UART_INVALID_VALUE, lIoctl );
+    }
```
Comment (cmcgcc) [inline libraries/abstractions/common_io/test/test_iot_uart.c]: Does it also need parity setup?

File: libraries/abstractions/common_io/test/test_iot_uart.c [hunk 4]
```diff
@@ -0,0 +1,817 @@
...
+
+    uint8_t cpBuffer[ testIotUART_BUFFER_LENGTH + 1 ] = { 0 };
+    uint8_t cpBufferRead[ testIotUART_BUFFER_LENGTH + 1 ] = { 0 };
+    uint8_t uStringCompare = 1;
+
+    strncpy( (char*)cpBuffer, testIotUART_BUFFER_STRING, testIotUART_BUFFER_LENGTH );
+
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
```
Comment (cmcgcc) [inline libraries/abstractions/common_io/test/test_iot_uart.c]: Could you keep Test protect? It was there in the original file.

File: libraries/abstractions/common_io/test/test_iot_uart.c (hunk 2 above)
Comment (lundinc2) [reply libraries/abstractions/common_io/test/test_iot_uart.c]: This is a rebase issue. We should not check any files in but the assisted test scripts. I will make a task of rebasing and reopening the PR.

File: libraries/abstractions/common_io/test/test_iot_uart.c (hunk 3 above)
Comment (lundinc2) [reply libraries/abstractions/common_io/test/test_iot_uart.c]: Same as above

File: libraries/abstractions/common_io/test/test_iot_uart.c (hunk 4 above)
Comment (lundinc2) [reply libraries/abstractions/common_io/test/test_iot_uart.c]: Same as above

**Comment by qiutongs** [comment]
I will copy your code and re-create a PR.

**Pull request closed without merging**